aws_lambda_powertools/utilities/idempotency/persistence/dynamodb.py

```diff
@@ -234,16 +235,51 @@ def _put_record(self, data_record: DataRecord) -> None:
                     ":now_in_millis": {"N": str(int(now.timestamp() * 1000))},
                     ":inprogress": {"S": STATUS_CONSTANTS["INPROGRESS"]},
                 },
+                **(
+                    {"ReturnValuesOnConditionCheckFailure": "ALL_OLD"}  # type: ignore
+                    if self.boto3_supports_condition_check_failure(boto3.__version__)
+                    else {}
+                ),
             )
         except ClientError as exc:
             error_code = exc.response.get("Error", {}).get("Code")
             if error_code == "ConditionalCheckFailedException":
+                old_data_record = self._item_to_data_record(exc.response["Item"]) if "Item" in exc.response else None
+                if old_data_record is not None:
+                    logger.debug(
+                        f"Failed to put record for already existing idempotency key: "
+                        f"{data_record.idempotency_key} with status: {old_data_record.status}, "
+                        f"expiry_timestamp: {old_data_record.expiry_timestamp}, "
+                        f"and in_progress_expiry_timestamp: {old_data_record.in_progress_expiry_timestamp}",
+                    )
+                    self._save_to_cache(data_record=old_data_record)
```

Comment:
Great catch! Since we are no longer calling [get_record](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/utilities/idempotency/persistence/base.py#L421) in case of returning from an exception, we would lose the ability to cache without it.